### PR TITLE
[BENCH-970] Fetch user email from sam using userRequest

### PR DESCRIPTION
**Pull request by dexamundsen**
- Fetch user email from sam using userRequest
- Check if aws is enabled for the user during clone workspace

### CODE REVIEW

**Review by rogerwangcs — APPROVED** [review]

File: service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java [hunk 1]
```diff
@@ -485,7 +489,9 @@ public ResponseEntity<ApiCreateCloudContextResult> createCloudContext(
 
     CloudPlatform cloudPlatform = CloudPlatform.fromApiCloudPlatform(apiCloudPlatform);
     if (cloudPlatform == CloudPlatform.AWS) {
-      featureService.featureEnabledCheck(FeatureService.AWS_ENABLED, userRequest.getEmail());
+      featureService.featureEnabledCheck(
+          FeatureService.AWS_ENABLED,
+          samService.getUserEmailFromSamAndRethrowOnInterrupt(userRequest));
```
Comment (yuhuyoyo) [inline service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java:494]: do we have a situation when the petSA create a workspace?

**Review by zloery — APPROVED** [review]

File: service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java (hunk 1 above)
Comment (dexamundsen) [reply service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java:494]: I checked the created_by_email field in devel and did not find any pet sa emails.

Comment (yuhuyoyo) [reply service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java:494]: then why do we need to query sam for user email?

Comment (dexamundsen) [reply service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java:494]: User Request may not have the mail address filled, if it only has a access token. email address is however needed for feature check

**Pull request merged**